File: backend/commons/security/security-entity/src/main/java/com/kraken/security/entity/owner/ApplicationOwner.java
package com.kraken.security.entity.owner;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;
import lombok.NonNull;
import lombok.Value;

@Value
@Builder(toBuilder = true)
public class ApplicationOwner implements Owner {

  String applicationId;

  @JsonCreator
  ApplicationOwner(
      @NonNull @JsonProperty("applicationId") final String applicationId
  ) {
    super();
    this.applicationId = applicationId;
  }

  @Override
  public OwnerType getType() {
    return OwnerType.APPLICATION;
  }
}

File: frontend/projects/security/src/lib/entities/application-owner.ts
import {Owner} from 'projects/security/src/lib/entities/owner';
import {OwnerType} from 'projects/security/src/lib/entities/owner-type';

export class ApplicationOwner implements Owner {
  constructor(
    public applicationId: string,
    public type: OwnerType = 'APPLICATION') {
  }
}
